Registration/RegisterWithAzure.psm1

```diff
@@ -206,15 +206,16 @@ function Set-AzsRegistration{
     Log-Output "*********************** Begin log: $($PSCmdlet.MyInvocation.MyCommand.Name) ***********************`r`n"
 
     Validate-ResourceGroupLocation -ResourceGroupLocation $ResourceGroupLocation
-    $azureAccountInfo = Get-AzureAccountInfo -AzureContext $AzureContext
+    Log-AzureAccountInfo -AzureContext $AzureContext
 
     try
     {
         $session = Initialize-PrivilegedEndpointSession -PrivilegedEndpoint $PrivilegedEndpoint -PrivilegedEndpointCredential $PrivilegedEndpointCredential -Verbose
         $stampInfo = Confirm-StampVersion -PSSession $session
 
         # Configure Azure Bridge
-        $servicePrincipal = New-ServicePrincipal -RefreshToken $azureAccountInfo.Token.RefreshToken -AzureEnvironmentName $AzureContext.Environment.Name -TenantId $azureAccountInfo.TenantId -PSSession $session
+        $refreshToken = (Get-AzToken -FromCache -Verbose).GetRefreshToken()
```

Comment:
Need to pass the context to this function; your function accepts target context as an input, so if context of UserB is passwed-in, this needs to get refresh token of user for that context, not user of "current context" which could be user UserA.

Reply:
thanks! will fix this.

Reply:
fixed.